src/mos_editor.c

```diff
@@ -240,6 +242,66 @@ UINT24 mos_EDITLINE(char * buffer, int bufferLength, UINT8 clear) {
 			case 0x87: {	// END
 				insertPos = gotoEditLineEnd(insertPos, len);
 			} break;
+			
+			case 0x9F: //F1
+			case 0xA0: //F2
+			case 0xA1: //F3
+			case 0xA2: //F4
+			case 0xA3: //F5
+			case 0xA4: //F6
+			case 0xA5: //F7
+			case 0xA6: //F8
+			case 0xA7: //F9
+			case 0xA8: //F10
+			case 0xA9: //F11	
+			case 0xAA: //F12
+			{
+				if (hotkey_strings[keyc - 159] != NULL) {
+
+					char *wildcardPos = strstr(hotkey_strings[keyc - 159], "%s");
+
+					if (wildcardPos == NULL) { //No wildcard in the hotkey string
+
+						removeEditLine(buffer, insertPos, len);
+						strcpy(buffer, hotkey_strings[keyc - 159]);
+						printf("%s", buffer);
+						len = strlen(buffer);
+						insertPos = len;
+						keya = 0x0D;
+					} else {
+
+						UINT8 prefixLength = wildcardPos - hotkey_strings[keyc - 159];
+						UINT8 replacementLength = strlen(buffer);
+						UINT8 suffixLength = strlen(wildcardPos + 2);
+						char *result;
+
+						if (prefixLength + replacementLength + suffixLength + 1 >= bufferLength) {
+							break;  // Exceeds max command length (256 chars)
+						}
+
+						result = malloc(prefixLength + replacementLength + suffixLength + 1); // +1 for null terminator
+
+						strncpy(result, hotkey_strings[keyc - 159], prefixLength); //Copy the portion preceding the wildcard to the buffer
+						result[prefixLength] = '\0'; //Terminate
+
+						strcat(result, buffer);
+						strcat(result, wildcardPos + 2);
+
+						removeEditLine(buffer, insertPos, len);
+						strcpy(buffer, result);
+						printf("%s", buffer);
+						len = strlen(buffer);
+						insertPos = len;
+						
+						keya = 0x0D;
+
+						free(result);
+
+					}
+
+				} else break;

```

Comment:
this feels wrong...  I'd expect this should actually be:
```suggestion
				} break;
```
as otherwise execution is just going to fall thru into the next "default" block

Reply:
This is because we want the program to fall through to the remainder of the switch block and be processed as if return had just been pressed (hence setting `keya` to 0x0D).